activemq-client/src/main/java/org/apache/activemq/ActiveMQConnection.java

```diff
@@ -1382,7 +1385,30 @@ private void doAsyncSendPacket(Command command) throws JMSException {
      * @throws JMSException
      */
     public void syncSendPacket(final Command command, final AsyncCallback onComplete) throws JMSException {
-        if(onComplete==null) {
+        CompletionListener completionListener = null;
+        if (onComplete != null) {
+            completionListener = new CompletionListener() {
+                @Override
+                public void onCompletion(jakarta.jms.Message message) {
+                    onComplete.onSuccess();
+                }
+
+                @Override
+                public void onException(jakarta.jms.Message message, Exception e) {}
```

Comment:
why not passing to the onComplete?

Reply:
Actually that's a good callout. Initially I looked at the AsyncCallback interface it doesn't have onException. But then it extends ExceptionListener so it does have it.